File: Algorithm/Java/src/main/java/com/yugutou/charpter12_string/level1/ToLowerCase.java
package com.yugutou.charpter12_string.level1;

public class ToLowerCase {

    public static void main(String[] args) {
        String s = "Hello";
        System.out.println(toLowerCase(s));
    }

    public static String toLowerCase(String s) {
        int n = s.length();
        char[] chars = s.toCharArray();
        for (int i = 0; i < n; ++i) {
            if (chars[i] >= 65 && chars[i] <= 90) {
                chars[i] += 32;
            }
        }
        String str = new String(chars);
        return str;
    }
}

File: Algorithm/Code_Materials/c_or_cpp/charpter12_string/level1/to_lower_case.cpp
#include <stdio.h>

char* toLowerCase(char* s) {
    int i = 0;
    while (s[i] != '\0') {  // 遍历字符串直到遇到结束符 \0
        if (s[i] >= 'A' && s[i] <= 'Z') {  // 判断字符是否是大写字母
            s[i] += 32;  // 转换为小写字母
        }
        i++;
    }
    return s;
}

int main() {
    char s[] = "Hello World";
    printf("%s\n", toLowerCase(s));
    return 0;
}